bundles/framework/mydata/service/ViewService.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * @class Oskari.mapframework.bundle.personaldata.service.ViewService
+ *
+ * Requests for a search to be made with the given query and provides
+ * callbacks
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.personaldata.service.ViewService',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     *
+     * @param {String}
+     *            viewUrl ajax URLfor view AJAX-operations
+     */
+
+    function (viewUrl) {
+        this._viewUrl = viewUrl;
+    }, {
+        /** @static @property __qname fully qualified name for service */
+        __qname: 'Oskari.mapframework.bundle.personaldata.service.ViewService',
+        /**
+         * @method getQName
+         * @return {String} fully qualified name for service
+         */
+        getQName: function () {
+            return this.__qname;
+        },
+        /** @static @property __name service name */
+        __name: 'ViewService',
+        /**
+         * @method getName
+         * @return {String} service name
+         */
+        getName: function () {
+            return this.__name;
+        },
+        /**
+         * @method loadViews
+         * Makes the actual ajax call to view service implementation
+         * @param {String}
+         *            type view type [USER | PUBLISHED]
+         * @param {Function}
+         *            callback callback method called when operation completes. The
+         * function will receive boolean value as parameter indicating success(true)
+         * or error(false). IF success, second parameter has the response.
+         */
+        loadViews: function (type, callback) {
+            var me = this;
+            if (!type) {
+                type = 'USER';
+            }
+
+            jQuery.ajax({
+                url: me._viewUrl + 'action_route=GetViews',
```

Comment:
Lets drop the configurable base url and just use:
```suggestion
                url: Oskari.urls.getRoute('GetViews'),
```